### PR TITLE
Feature/5322 put description

### PR DESCRIPTION
**Pull request by RuthBerry96**
[AB#6740](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6740)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace NHSD.BuyingCatalogue.Ordering.Domain.Common
+{
+    public abstract class ValueObject
```
Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:6]: should probably have some unit tests

Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:6]: a summary on these classes to explain what they're for would be nice

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+namespace NHSD.BuyingCatalogue.Ordering.Domain.Common
+{
+    public abstract class ValueObject
+    {
+        /// <summary>
+        ///     Gets a list of values of this instance to be used for equality.
+        /// </summary>
+        /// <returns>A list of values of this instance to be used for equality.</returns>
+        protected abstract IEnumerable<object> GetEqualityComponents();
+        /// <summary>
```
Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:13]: whitespace!

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            return GetEqualityComponents()
+                .Aggregate(1, (current, obj) =>
+                {
+                    unchecked
+                    {
+                        return current * 23 + (obj?.GetHashCode() ?? 0);
+                    }
+                });
+        }
+        /// <summary>
```
Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:45]: whitespace!

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                return true;
+            }
+            if (ReferenceEquals(a, null) || ReferenceEquals(b, null))
+            {
+                return false;
+            }
+            return a.Equals(b);
+        }
+        /// <summary>
```
Comment (CallumBradburyBJSS) [inline src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:63]: … … whitespace! xD

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description 5322.feature [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Scenario: 4. Updating an order, with a description, exceeding it's maximum limit, returns a relevant error message
+    When the user makes a request to update the description with the ID C000014-01
+        | Description              |
+        | #A string of length 101# |
+    Then a response with status code 400 is returned
+    And the response contains the following errors
+        | Id                      | Field       |
+        | OrderDescriptionTooLong | Description |
+
+@ignore
```
Comment (CallumBradburyBJSS) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description 5322.feature:48]: If you get latest, you can follow the approach on the other test files to un-ignore these

File: NHSD.BuyingCatalogue.Ordering.Api.sln [hunk 6]
```diff
@@ -1,96 +1,103 @@
-﻿
```
Comment (CallumBradburyBJSS) [inline NHSD.BuyingCatalogue.Ordering.Api.sln:1]: I wonder what caused this to replace the entire file

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs (hunk 2 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:13]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs (hunk 3 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:45]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs (hunk 4 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:63]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description 5322.feature (hunk 5 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description 5322.feature:48]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs (hunk 1 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:6]: 👍

Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.Domain/Common/ValueObject.cs:6]: Taken the unit tests for Buying Catalogue

File: NHSD.BuyingCatalogue.Ordering.Api.sln (hunk 6 above)
Comment (RuthBerry96) [reply NHSD.BuyingCatalogue.Ordering.Api.sln:1]: Who knows, what would you like me to do?

**Pull request merged**